### PR TITLE
Replace deepcopy with copy.

### PR DESCRIPTION
**Pull request by wang2yn84**
# Description

This operations takes 3ms for us, with copy() and it's reduced to <0.4ms. The deepcopy is more expensive than copy. This dictionary is {str, int}, therefore copy is enough.

The rest of the description includes relevant details and context, examples:
- why is this change being made,
- the problem being solved and any relevant context,
- why this is a good solution,
- some information about the specific implementation,
- shortcomings of the solution and possible future improvements.

If the change fixes a Github issue, please include a link, e.g.,:
FIXES: #123456

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/runner/tpu_runner.py [hunk 1]
```diff
@@ -834,7 +834,7 @@ def _execute_model(
 
                 return ModelRunnerOutput(
                     req_ids=self.input_batch.req_ids,
-                    req_id_to_index=self.input_batch.req_id_to_index,
+                    req_id_to_index=self.input_batch.req_id_to_index.copy(),
```
Comment (wenxindongwork) [inline tpu_inference/runner/tpu_runner.py]: why is the `copy()` needed here?

File: tpu_inference/runner/tpu_runner.py [hunk 2]
```diff
@@ -1063,7 +1063,7 @@ def _sample_from_logits(
 
         model_runner_output = ModelRunnerOutput(
             req_ids=req_ids,
-            req_id_to_index=self.input_batch.req_id_to_index,
+            req_id_to_index=self.input_batch.req_id_to_index.copy(),
```
Comment (wenxindongwork) [inline tpu_inference/runner/tpu_runner.py]: and similarly, why is the copy needed here?

File: tpu_inference/runner/tpu_runner.py (hunk 1 above)
Comment (kyuyeunk) [reply tpu_inference/runner/tpu_runner.py]: quote from @yuyanpeng-google:

> It's for async scheduler. Since the next step would change the input_batch, it need deepcopy to prevent race condition. When the previous result is ready to get, it need the req_id_to_index(of the previous batch)

Comment (yuyanpeng-google) [reply tpu_inference/runner/tpu_runner.py]: I think this one is not for the AsyncTPUModelRunnerOutput, and we don't need the copy here.

File: tpu_inference/runner/tpu_runner.py (hunk 2 above)
Comment (yuyanpeng-google) [reply tpu_inference/runner/tpu_runner.py]: I thought it's not necessary for not AsyncTPUModelRunnerOutput.

File: tpu_inference/runner/tpu_runner.py (hunk 1 above)
Comment (wang2yn84) [reply tpu_inference/runner/tpu_runner.py]: Thank you for the comments, thought it's also used for async before. Removed!

File: tpu_inference/runner/tpu_runner.py (hunk 2 above)
Comment (wang2yn84) [reply tpu_inference/runner/tpu_runner.py]: Also removed.

**Review by wenxindongwork — APPROVED** [review]
